applications/vanilla/views/categories/discussions.php

```diff
@@ -34,12 +36,15 @@
                         </div>
                     <?php endif; ?>
 
-                </div>
-
-            <?php endif; ?>
+                <?php else: ?>
+                    <div class="Empty"><?php echo t('No discussions were found.'); ?></div>
+                <?php endif; ?>
+            </div>
 
         <?php endforeach; ?>
-    <?php else: ?>
-        <div class="Empty"><?php echo t('No categories were found.'); ?></div>
+    <?php else:
```

Comment:
This seems wrong. You are outside of your loop. It seems like you still need that "No categories were found" from above.